pkg/ck8s/workload_cluster.go

```diff
@@ -300,6 +355,58 @@ func (w *Workload) doK8sdRequest(ctx context.Context, method, endpoint string, r
 	return nil
 }
 
+func (w *Workload) doK8sdRequestForMachine(ctx context.Context, machine *clusterv1.Machine, method, endpoint string, header map[string][]string, request any, response any) error {
```

Comment:
Hm, maybe refactor the already existing `doK8sdRequest` instead of duplicating the logic here?